File: app/src/main/java/vn/loitp/app/activity/customviews/dialog/swipeawaydialog/SADilog.java
package vn.loitp.app.activity.customviews.dialog.swipeawaydialog;

import android.app.AlertDialog;
import android.app.Dialog;
import android.app.ProgressDialog;
import android.graphics.Color;
import android.graphics.drawable.ColorDrawable;
import android.os.AsyncTask;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.Button;

import loitp.basemaster.R;
import vn.loitp.core.common.Constants;
import vn.loitp.core.utilities.LDialogUtil;
import vn.loitp.core.utilities.LLog;
import vn.loitp.utils.util.ToastUtils;
import vn.loitp.views.LToast;
import vn.loitp.views.dialog.swipeawaydialog.support.v4.SwipeAwayDialogFragment;

public class SADilog extends SwipeAwayDialogFragment {
    private final String TAG = getClass().getSimpleName();
    public static String KEY;
    public final static int KEY_1 = 1;
    public final static int KEY_2 = 2;
    public final static int KEY_3 = 3;
    public final static int KEY_4 = 4;
    public final static int KEY_5 = 5;
    public final static int KEY_6 = 6;

    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        Bundle bundle = getArguments();
        int key = Constants.INSTANCE.getNOT_FOUND();
        if (bundle != null) {
            key = bundle.getInt(KEY);
        }
        LLog.d(TAG, "key " + key);
        switch (key) {
            case KEY_1:
                return show1();
            case KEY_2:
                return show2();
            case KEY_3:
                return show3();
            case KEY_4:
                return showList();
            case KEY_5:
                return showProgress();
            case KEY_6:
                return showCustom();
            default:
                return show1();
        }
    }

    private AlertDialog show1() {
        return LDialogUtil.showDialog1(getActivity(), "Title", "Msg", "Button 1", new LDialogUtil.Callback1() {
            @Override
            public void onClick1() {
                ToastUtils.showShort("Click 1");
            }
        });
    }

    private AlertDialog show2() {
        return LDialogUtil.showDialog2(getActivity(), "Title", "Msg", "Button 1", "Button 2", new LDialogUtil.Callback2() {
            @Override
            public void onClick1() {
                ToastUtils.showShort("Click 1");
            }

            @Override
            public void onClick2() {
                ToastUtils.showShort("Click 2");
            }
        });
    }

    private AlertDialog show3() {
        return LDialogUtil.showDialog3(getActivity(), "Title", "Msg", "Button 1", "Button 2", "Button 3", new LDialogUtil.Callback3() {
            @Override
            public void onClick1() {
                ToastUtils.showShort("Click 1");
            }

            @Override
            public void onClick2() {
                ToastUtils.showShort("Click 2");
            }

            @Override
            public void onClick3() {
                ToastUtils.showShort("Click 3");
            }
        });
    }

    private AlertDialog showList() {
        int size = 50;
        String arr[] = new String[size];
        for (int i = 0; i < size; i++) {
            arr[i] = "Item " + i;
        }
        return LDialogUtil.showDialogList(getActivity(), "Title", arr, new LDialogUtil.CallbackList() {
            @Override
            public void onClick(int position) {
                ToastUtils.showShort("Click position " + position + ", item: " + arr[position]);
            }
        });
    }

    private AlertDialog showProgress() {
        ProgressDialog progressDialog = LDialogUtil.showProgressDialog(getActivity(), 100, "Title", "Message", false, ProgressDialog.STYLE_HORIZONTAL, null, null);
        new AsyncTask<Void, Integer, Void>() {
            int i = 0;

            @Override
            protected void onPreExecute() {
                super.onPreExecute();
            }

            @Override
            protected Void doInBackground(Void... voids) {
                for (i = 0; i < progressDialog.getMax(); i++) {
                    publishProgress(i);
                    try {
                        Thread.sleep(100);
                    } catch (InterruptedException e) {
                        e.printStackTrace();
                    }
                }
                return null;
            }

            @Override
            protected void onProgressUpdate(Integer... values) {
                super.onProgressUpdate(values);
                progressDialog.setProgress(values[0]);

            }

            @Override
            protected void onPostExecute(Void aVoid) {
                progressDialog.dismiss();
                super.onPostExecute(aVoid);
            }
        }.execute();
        return progressDialog;
    }

    private AlertDialog showCustom() {
        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
        LayoutInflater inflater = getActivity().getLayoutInflater();
        View view = inflater.inflate(R.layout.dlg_swipe_away_custom, null);

        Button btY = (Button) view.findViewById(R.id.bt_y);
        btY.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                LToast.show(getActivity(), "Click yes");
            }
        });

        Button btN = (Button) view.findViewById(R.id.bt_n);
        btN.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                LToast.show(getActivity(), "Click no");
            }
        });

        builder.setView(view);
        AlertDialog dialog = builder.create();

        dialog.getWindow().setBackgroundDrawable(new ColorDrawable(Color.TRANSPARENT));
        return dialog;
    }
}

File: app/src/main/java/vn/loitp/app/activity/customviews/dialog/swipeawaydialog/SADilog.kt
package vn.loitp.app.activity.customviews.dialog.swipeawaydialog

import android.annotation.SuppressLint
import android.app.AlertDialog
import android.app.Dialog
import android.app.ProgressDialog
import android.graphics.Color
import android.graphics.drawable.ColorDrawable
import android.os.AsyncTask
import android.os.Bundle
import android.widget.Button
import com.core.common.Constants
import com.core.utilities.LDialogUtil
import com.core.utilities.LLog
import com.views.LToast
import com.views.dialog.swipeawaydialog.support.SwipeAwayDialogFragment
import loitp.basemaster.R

class SADilog : SwipeAwayDialogFragment() {
    private val TAG = javaClass.simpleName

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val bundle = arguments
        var key = Constants.NOT_FOUND
        if (bundle != null) {
            key = bundle.getInt(KEY)
        }
        LLog.d(TAG, "key $key")
        return when (key) {
            KEY_1 -> show1()
            KEY_2 -> show2()
            KEY_3 -> show3()
            KEY_4 -> showList()
            KEY_5 -> showProgress()
            KEY_6 -> showCustom()
            else -> show1()
        }
    }

    private fun show1(): AlertDialog {
        return LDialogUtil.showDialog1(context!!, "Title", "Msg", "Button 1",
                object : LDialogUtil.Callback1 {
                    override fun onClick1() {
                        LToast.show(context!!, "Click 1", R.drawable.l_bkg_horizontal)
                    }
                })
    }

    private fun show2(): AlertDialog {
        return LDialogUtil.showDialog2(context!!, "Title", "Msg", "Button 1", "Button 2", object : LDialogUtil.Callback2 {
            override fun onClick1() {
                LToast.showShort(context!!, "Click 1", R.drawable.l_bkg_horizontal)
            }

            override fun onClick2() {
                LToast.showShort(context!!, "Click 2", R.drawable.l_bkg_horizontal)
            }
        })
    }

    private fun show3(): AlertDialog {
        return LDialogUtil.showDialog3(context!!, "Title", "Msg", "Button 1", "Button 2", "Button 3", object : LDialogUtil.Callback3 {
            override fun onClick1() {
                LToast.showShort(context!!, "Click 1", R.drawable.l_bkg_horizontal)
            }

            override fun onClick2() {
                LToast.showShort(context!!, "Click 2", R.drawable.l_bkg_horizontal)
            }

            override fun onClick3() {
                LToast.showShort(context!!, "Click 3", R.drawable.l_bkg_horizontal)
            }
        })
    }

    private fun showList(): AlertDialog {
        val size = 50
        val arr = arrayOfNulls<String>(size)
        for (i in 0 until size) {
            arr[i] = "Item $i"
        }
        return LDialogUtil.showDialogList(context!!, "Title", arr, object : LDialogUtil.CallbackList {
            override fun onClick(position: Int) {
                LToast.show(context!!, "Click position " + position + ", item: " + arr[position], R.drawable.l_bkg_horizontal)
            }
        })
    }

    private fun showProgress(): AlertDialog {
        val progressDialog = LDialogUtil.showProgressDialog(activity!!, 100, "Title", "Message", false, ProgressDialog.STYLE_HORIZONTAL, null, null)
        //TODO convert asynctask to rx
        object : AsyncTask<Void, Int, Void>() {
            var i = 0

            override fun doInBackground(vararg voids: Void): Void? {
                i = 0
                while (i < progressDialog.max) {
                    publishProgress(i)
                    try {
                        Thread.sleep(100)
                    } catch (e: InterruptedException) {
                        e.printStackTrace()
                    }

                    i++
                }
                return null
            }

            protected override fun onProgressUpdate(vararg values: Int?) {
                super.onProgressUpdate(*values)
                values[0]?.let {
                    progressDialog.progress = it
                }

            }

            override fun onPostExecute(aVoid: Void) {
                progressDialog.dismiss()
                super.onPostExecute(aVoid)
            }
        }.execute()
        return progressDialog
    }

    @SuppressLint("InflateParams")
    private fun showCustom(): AlertDialog {
        val builder = AlertDialog.Builder(activity)
        val inflater = activity!!.layoutInflater
        val view = inflater.inflate(R.layout.dlg_swipe_away_custom, null)

        val btY = view.findViewById<Button>(R.id.bt_y)
        btY.setOnClickListener { LToast.show(activity!!, "Click yes") }

        val btN = view.findViewById<Button>(R.id.bt_n)
        btN.setOnClickListener { LToast.show(activity!!, "Click no") }

        builder.setView(view)
        val dialog = builder.create()
        if (dialog.window != null) {
            dialog.window!!.setBackgroundDrawable(ColorDrawable(Color.TRANSPARENT))
        }
        return dialog
    }

    companion object {
        val KEY: String? = null
        const val KEY_1 = 1
        const val KEY_2 = 2
        const val KEY_3 = 3
        const val KEY_4 = 4
        const val KEY_5 = 5
        const val KEY_6 = 6
    }
}
